### ISSUE TITLE
Regression: wrong laser color 

### ISSUE THREAD

**Issue by derhass** [dxx-rebirth/dxx-rebirth#416]
### Environment
Self compiled, git commit `3a6bfb8dd621fe469064db5379151f15fdae967f`

#### Operating System Environment

* Microsoft Windows
  * [ ] Microsoft Windows XP
  * [ ] Microsoft Windows Vista
  * [ ] Microsoft Windows 7
  * [ ] Microsoft Windows 8
  * [ ] Microsoft Windows 8.1
  * [ ] Microsoft Windows 10

* Mac OS X
  * [ ] Mac OS X 10.8
  * [ ] Mac OS X 10.9
  * [ ] Mac OS X 10.10
  * [ ] Mac OS X 10.11
  * [ ] Mac OS X 10.12
  * [ ] Mac OS X 10.13

* Debian
  * [ ] Debian Wheezy
  * [ ] Debian Jessie
  * [x] Debian Stretch
  * [ ] Debian Buster
* Fedora
  * [ ] Fedora 26
  * [ ] Fedora 27
  * [ ] Fedora 28
  * [ ] Fedora 29
  * [ ] Rawhide
* Ubuntu
  * [ ] Ubuntu 16.04 LTS (Xenial Xerus)
  * [ ] Ubuntu 16.10 (Yakkety Yak)
  * [ ] Ubuntu 17.04 (Zesty Zapus)
  * [ ] Ubuntu 17.10 (Artful Aardvark)
  * [ ] Ubuntu 18.04 LTS (Bionic Beaver)
  * [ ] Ubuntu 18.10 (Cosmic Cuttlefish)

* [ ] Arch
* [ ] Gentoo

* [ ] OpenBSD
* [ ] FreeBSD

#### CPU environment

* [ ] x86 (32-bit Intel/AMD)
* [x] x86\_64 (64-bit Intel/AMD)
* [ ] ARM (32-bit)
* [ ] ARM64 (64-bit; sometimes called AArch64)

#### Game environment
I've observed this with the Descent: First Strike PC game data.

### Description

Laser level 1 appears blue and green instead of red/orange. When dynamic is lighting is enabled, the dynamic light is also green:

![greenlaser](https://user-images.githubusercontent.com/1835702/54577063-9a3f4480-49fa-11e9-9801-44c59023e4d6.png)

#### Regression status

I bisected the issue and ended up with:
```
42a2e3ab0b420c3c7c4b7c3791525c0b15352462 is the first bad commit
commit 42a2e3ab0b420c3c7c4b7c3791525c0b15352462
Author: Kp <kp@valhallalegends.com>
Date:   Wed Mar 6 05:03:48 2019 +0000

    Avoid crash loading polymodels with invalid subcalls

    Truncate the model at the first error.  This allows the level to load,
    but makes some or all of the model invisible.

:040000 040000 000bdab2a6f265da715f1b91df6c2b452ddf0679 879a11ee42de81af1db8f09672ce3de9d238e80f M      common
:040000 040000 2567397ac68b5873d6c2e7727281ad806211270c a835df167fb8ed315fec8d730a4111d3c39aa662 M      d1x-rebirth
:040000 040000 e3edc53570d2f2e2f93eae01de7d5851e303e0b3 af1750c8776c55b35d79a49d314d57be4a030f7b M      similar
```

### Steps to Reproduce

* Engines affected:
  * [x] D1X-Rebirth
  * [ ] D2X-Rebirth

it does not happen with `d2x-rebirth` even when playing the D1 campaign.

* Game modes affected:
  * all

**Comment by vLKp** [comment]
Confirmed.  In Descent 1, `g3_init_polygon_model` mangles colors as it traverses the model.  In Descent 2, it does not.  I changed Descent 1 to have a new call to `g3_init_polygon_model` at model load so that it could validate offsets used for subcalls.  This caused the regression by activating the color mangling code on models that were previously not mangled.  Affected models need to avoid ever remapping their colors, so it is not sufficient to make the Descent 1 polygon interpreter more like the Descent2 interpreter.
